ddht/v5_1/network.py

```diff
@@ -230,11 +230,11 @@ async def worker(
 
     async def _monitor_done(send_channel: trio.abc.SendChannel[ENRAPI]) -> None:
         async with send_channel:
-            while True:
-                # this `fail_after` is a failsafe to prevent deadlock situations
-                # which are possible with `Condition` objects.
-                with trio.fail_after(60):
-                    async with condition:
+            async with condition:
+                while True:
+                    # this `fail_after` is a failsafe to prevent deadlock situations
+                    # which are possible with `Condition` objects.
+                    with trio.fail_after(60):
```

Comment:
I've been just kicking the CI tests over and over to re-run them to try and find the failures.  So far this PR hasn't had a single timeout.  Not conclusive at all but... I think this is where the flakyness was coming from.

TLDR `Condition` objects seem to be the right tool for certain concurrency situations but they are also easy to use incorrectly (and potentially end up with deadlocks).